### PR TITLE
Gantt

### PR DESCRIPTION
**Pull request by MOHITRAJANI**
Functionality Added:
1) Gantt Visualization for the Jobs in the current History.
    As we know the Jobs have a start time and end time as returned by the Job Metrics, We used the same to map tasks on a 
    Gantt Chart using a Library called Frappe-Gantt

2) There are many features that Frappe-Gantt doesn't possess but we needed them for our project and hence I have implemented them in the library, these features are:
  - Hour and Minute view also added to the library
  - Header row shows Date-Month-Hour-Seconds
  - A new SVG (Diamond shaped) is added which gets shown whenever the task is too small for the Viewing-scale that is being used, eg. If a Hour or Week view is used and the tasks are only in seconds then you'd see a diamond instead of a bar.

3) Because we want to keep the changes that we made in the npm package frappe-gantt @shubham4315 found a nice package called patch-package which keeps the changes of the package and restores these changes as a post-install.

4) For the data part, I am getting the Current History ID from which I fetch all the jobs in the history and then fetch their Start and End time from the jobs/metrics api.

5) This is how you navigate to this feature:
Screenshot from 2022-11-13 14-58-14

6) How a single task looks on the timeline
Screenshot from 2022-11-13 14-57-28

7) Diamond svg when the tasks are too small for the scale
Screenshot from 2022-11-13 15-03-45

8) All the scales available for the Timeline
Screenshot from 2022-11-13 15-04-23

9) The popover looks even more prettier and readable now, Just some basic CSS.
Screenshot from 2022-11-21 20-39-18

This is still at a very early stage and many improvements are still upcoming. 

The tasks for this feature will be performed my Me and @shubham4315  as a part of our Masters Project.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Manual testing is still not added for the functionality and shall be added in the later stages of deployment
  

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/patches/frappe-gantt+0.6.1.patch [hunk 1]
```diff
@@ -0,0 +1,2537 @@
+diff --git a/node_modules/frappe-gantt/src/bar.js b/node_modules/frappe-gantt/src/bar.js
```
Comment (dannon) [inline client/patches/frappe-gantt+0.6.1.patch]: As an initial comment, we won't want to patch dependencies this way with such large changes -- I would recommend forking the project and pushing it to npm with the changes you need as a separate project, or even using a git dependency like:
```
  "dependencies": {
    "frappe-gantt": "github:yourgithub/frappe-gantt#123kj23kj",
```

**Comment by dannon** [comment]
Looks pretty cool as a start, excited to see where this goes!

File: client/patches/frappe-gantt+0.6.1.patch (hunk 1 above)
Comment (MOHITRAJANI) [reply client/patches/frappe-gantt+0.6.1.patch]: Cool, will work on it then, thanks for this input :)

Comment (MOHITRAJANI) [reply client/patches/frappe-gantt+0.6.1.patch]: Git dependency for the package is now added to the feature :100:

**Comment by itisAliRH** [comment]
@jdavcs This pr was a draft and was not ready to merge. Could you please roll it back?

**Comment by jdavcs** [comment]
@itisAliRH Sorry, that was a mistake - I never intended to merge this! Will roll back in a min.

**Comment by jdavcs** [comment]
So, here's what happened. [This force-push](https://github.com/galaxyproject/galaxy/compare/3e074307009c9d7da874a06f4052e1dc514e19e9..dc59759ad4b6344641d31a957920437f7f76f7f7) resulted in the Gantt branch being reset to the current state of the galaxyproject dev branch. The next merge on the dev branch (which was mine: #14993: just click the commit f3b109a) resulted in this PR being marked as merged. As per @nsoranzo : "if a PR contains only commits that become part of its target branch when another PR is merged, the PR is "merged" as well by GitHub. But no new commit is added to the upstream branch, the PR is simply closed as "merged"." This also explains why this PR has no commits. The old commits state can be recovered here https://github.com/galaxyproject/galaxy/commit/3e074307009c9d7da874a06f4052e1dc514e19e9

**Pull request merged**